### PR TITLE
Designing Metrics Blog

### PR DESCRIPTION
**Pull request by callum-mcdata**
## Description & motivation
Adds the how to design metrics blog

## Checklist
If you added new pages (delete if not applicable):
- [x] The new page has a unique filename

### CODE REVIEW

File: website/blog/2022-10-18-how-to-design-metrics.md [hunk 1]
```diff
@@ -0,0 +1,367 @@
...
+
+In past roles, I’ve been referred to as the `dbt zealot` and I’ll own that title! But it’s not a surprise - dbt was built to serve analytics engineers exactly like me. It gave us flexibility and power to serve our organizations. But it always felt like dbt best served **us.**
+
+The Semantic Layer takes this to the next level **by coupling dbt’s mature data modeling framework with semantic definitions.** The result is a first of its kind data experience that serves both the analytics engineers and stakeholders. Metrics are the first step towards this vision, allowing users to version control and centrally define their key business metrics in a single repo **while also** serving them to the entire business.
+
+However, this is still a relatively new part of the dbt tool box and you probably have a lot of questions on how **exactly** you can do that. Contained within this blog are our early recommendations for how to design and structure metrics, along with examples so that you can apply the same principles to your use cases. They were developed by combining the overall philosophy of dbt with our hands on learning gathered during the beta period and internal testing.
```
Comment (mirnawong1) [inline website/blog/2022-10-18-how-to-design-metrics.md]:
```suggestion
However, this is still a relatively new part of the dbt toolbox and you probably have a lot of questions on how **exactly** you can do that. This blog contains our early recommendations for how to design and structure metrics, along with examples so that you can apply the same principles to your use cases. We developed these recommendations by combining the overall philosophy of dbt, with our hands-on learning gathered during the beta period and internal testing.
```

File: website/blog/2022-10-18-how-to-design-metrics.md [hunk 2]
```diff
@@ -0,0 +1,367 @@
...
+
+## How to design your metrics
+
+**Pre-reading:** We recommend reading through the [metrics documentation](https://docs.getdbt.com/docs/building-a-dbt-project/metrics), which contains a table of all the required/optional properties.
```
Comment (mirnawong1) [inline website/blog/2022-10-18-how-to-design-metrics.md]:
```suggestion
**Pre-reading:** We recommend reading through the [metrics documentation](/docs/building-a-dbt-project/metrics), which contains a table of all the required/optional properties. 
```

File: website/blog/2022-10-18-how-to-design-metrics.md [hunk 3]
```diff
@@ -0,0 +1,367 @@
+---
+title: "How to design metrics: deciding what to model in the modeling layer vs the semantic layer"
```
Comment (jasnonaz) [inline website/blog/2022-10-18-how-to-design-metrics.md]: Love the way we incorporated the thesis in the title! This is a bit wordy - perhaps something along the lines of "How to design metrics: deciding which logic lives modeling layer vs the semantic layer"

File: website/blog/2022-10-18-how-to-design-metrics.md [hunk 4]
```diff
@@ -0,0 +1,367 @@
...
+description: "placeholder"
```
Comment (jasnonaz) [inline website/blog/2022-10-18-how-to-design-metrics.md]: todo: get pithy description of this blog

File: website/blog/2022-10-18-how-to-design-metrics.md [hunk 5]
```diff
@@ -0,0 +1,367 @@
...
+
+---
+
+## The power of a semantic layer on top of a mature data modeling framework
+
+As a longtime dbt Community member, I knew I had to get involved when I first saw the dbt Semantic Layer. It gave me a vision of a world where metrics and business logic were unified across an entire organization; a world where the data team was no longer bound to a single consuming experience and could enable their stakeholders in dozens of different ways. To me, it felt like the opportunity to contribute to the next step of what dbt could become.
```
Comment (jasnonaz) [inline website/blog/2022-10-18-how-to-design-metrics.md]: Do you remember the exact moment? Drew's talk at Coalesce? something else?

File: website/blog/2022-10-18-how-to-design-metrics.md [hunk 6]
```diff
@@ -0,0 +1,367 @@
...
+
+In past roles, I’ve been referred to as the `dbt zealot` and I’ll own that title! But it’s not a surprise - dbt was built to serve analytics engineers exactly like me. It gave us flexibility and power to serve our organizations. But it always felt like dbt best served **us.**
```
Comment (jasnonaz) [inline website/blog/2022-10-18-how-to-design-metrics.md]:
```SUGGESTION
In past roles, I’ve been referred to as the `dbt zealot` and I’ll gladly own that title as dbt was built to allow data practitioners like me to greatly expand the power of our work by writing code based off of software engineering principles. It gave us flexibility and power to serve our organizations. But I always wondered if there were more folks who could directly benefit from interacting with dbt.

File: website/blog/2022-10-18-how-to-design-metrics.md [hunk 7]
```diff
@@ -0,0 +1,367 @@
...
+
+The Semantic Layer takes this to the next level **by coupling dbt’s mature data modeling framework with semantic definitions.** The result is a first of its kind data experience that serves both the analytics engineers and stakeholders. Metrics are the first step towards this vision, allowing users to version control and centrally define their key business metrics in a single repo **while also** serving them to the entire business.
```
Comment (jasnonaz) [inline website/blog/2022-10-18-how-to-design-metrics.md]:
```SUGGESTION 
The Semantic Layer takes expands the reach of dbt **by coupling dbt’s mature data modeling framework with semantic definitions.** The result is a first of its kind data experience that serves both the data practitioners writing your analytics code and stakeholders who depend on it. Metrics are the first step towards this vision, allowing users to version control and centrally define their key business metrics in a single repo **while also** serving them to the entire business.

File: website/blog/2022-10-18-how-to-design-metrics.md [hunk 8]
```diff
@@ -0,0 +1,367 @@
...
+
+However, this is still a relatively new part of the dbt tool box and you probably have a lot of questions on how **exactly** you can do that. Contained within this blog are our early recommendations for how to design and structure metrics, along with examples so that you can apply the same principles to your use cases. They were developed by combining the overall philosophy of dbt with our hands on learning gathered during the beta period and internal testing.
+
+## How to design your metrics
+
+**Pre-reading:** We recommend reading through the [metrics documentation](/docs/building-a-dbt-project/metrics), which contains a table of all the required/optional properties. 
+
+### When to put business logic in the semantic layer vs the modeling layer
+
+Our instinct when designing metrics might be to encode as much as possible into the semantic layer. An example of this is case statements - the analytics engineer’s gut instinct might be to mimic tools of the past and provide complicated case statements for the metric `expression` property to try and capture the nuance of how it should be calculated.
```
Comment (jasnonaz) [inline website/blog/2022-10-18-how-to-design-metrics.md]: As much what - information?

File: website/blog/2022-10-18-how-to-design-metrics.md [hunk 9]
```diff
@@ -0,0 +1,367 @@
...
+
+## How to design your metrics
+
+**Pre-reading:** We recommend reading through the [metrics documentation](/docs/building-a-dbt-project/metrics), which contains a table of all the required/optional properties. 
+
+### When to put business logic in the semantic layer vs the modeling layer
+
+Our instinct when designing metrics might be to encode as much as possible into the semantic layer. An example of this is case statements - the analytics engineer’s gut instinct might be to mimic tools of the past and provide complicated case statements for the metric `expression` property to try and capture the nuance of how it should be calculated. 
+
+With the coupling of dbt’s modeling framework, this isn’t necessarily the right answer. In situations with more complicated logic, it might make more sense to keep this logic contained within the model and then reference it in the semantic layer.
```
Comment (jasnonaz) [inline website/blog/2022-10-18-how-to-design-metrics.md]:
```SUGGESTION 
But remember - you always have the option of performing this logic _in the modeling layer_. This is the key difference between dbt and other semantic layer offerings - by sitting the semantic layer atop a mature transformation layer, you always have the option to configure and optimize your logic within your models and then _define semantic components with intentionality_.

File: website/blog/2022-10-18-how-to-design-metrics.md [hunk 10]
```diff
@@ -0,0 +1,367 @@
...
+
+Getting the balance right has been a learning experience for our team as well. It’s easy to fall into comfortable patterns that might not be the best solution - it’s important for us to take a step back and think from first principles about the line of separation between what should live in a model and what should live in a metric. To explore this, we’ll walk through two examples of handling this divide.
```
Comment (jasnonaz) [inline website/blog/2022-10-18-how-to-design-metrics.md]:
```SUGGESTION 
Getting the balance  just right is a learning experience and developing community best practices and standards will take time, which is why it’s important for us to think from first principles. What should really be our goal when determining whether logic lives in a model or a metrics? 

To explore this question and begin to develop an intuition, we’ll walk through two examples of handling this divide.

File: website/blog/2022-10-18-how-to-design-metrics.md [hunk 11]
```diff
@@ -0,0 +1,367 @@
...
+
+### Basic example - designing your metrics
+
+In this example, we’ll cover the basics of defining a metric and a fairly straightforward example of where users can draw the line between the semantic layer and the modeling layer. You should finish this section with a better understanding of dbt metrics and its relationship to the modeling layer.
+
+We begin by taking a look at our end-state marts model called `order_events` that looks something like the below table, but on the order of millions of rows instead of five. Our finance team uses the below model to better understand revenue but inconsistencies in how its reported have led to requests that the data team centralize the definition in the dbt repo.
```
Comment (jasnonaz) [inline website/blog/2022-10-18-how-to-design-metrics.md]: Can we give a sense on _why_ we begin with this mart. Something like "In the past the mart was often your end stage layer before data was consumed in another tool or system. Now, the mart is our springboard for the creation of our metric"

File: website/blog/2022-10-18-how-to-design-metrics.md [hunk 12]
```diff
@@ -0,0 +1,367 @@
...
+
+### When to put business logic in the semantic layer vs the modeling layer
+
+Our instinct when designing metrics might be to encode as much as possible into the semantic layer. An example of this is case statements - the analytics engineer’s gut instinct might be to mimic tools of the past and provide complicated case statements for the metric `expression` property to try and capture the nuance of how it should be calculated. 
+
+With the coupling of dbt’s modeling framework, this isn’t necessarily the right answer. In situations with more complicated logic, it might make more sense to keep this logic contained within the model and then reference it in the semantic layer.
+
+Getting the balance right has been a learning experience for our team as well. It’s easy to fall into comfortable patterns that might not be the best solution - it’s important for us to take a step back and think from first principles about the line of separation between what should live in a model and what should live in a metric. To explore this, we’ll walk through two examples of handling this divide.
+
+### Basic example - designing your metrics
```
Comment (jasnonaz) [inline website/blog/2022-10-18-how-to-design-metrics.md:43]: I think we can tweak the headings here to make the sectioning clearer by tweaking our H2s and H3s.

## The power of a semantic layer on top of a mature data modeling framework
### 
When to put business logic in the semantic layer vs the modeling layer

## Basic example - revenue
### Designing your metric
### logic in the modeling layer vs the semantic layer
### choosing which dimensions to include on the metric

## advanced example - NPS
### designing a complex metric
### option 1 - keeping everything in the semantic layer
### option 2 - shift logic left into the modelling layer

## How to structure your metrics
### Folder structure 
### File structure 

## A call to action

File: website/blog/2022-10-18-how-to-design-metrics.md [hunk 13]
```diff
@@ -0,0 +1,367 @@
...
+
+**Metrics are important business objects unto themselves and should live separate from the model definitions.**
+
+## Looking To The Future
+
+This is just the beginning of dbt Metrics and the Semantic Layer. We have a number of exciting ideas for expanding capabilities that we plan to begin work on in the coming months. However, we can’t do that without you. 
+
+If you’ve gotten this far on a technical document on how to design and structure metrics then I’m going to assume you’re a member of the dbt community. Y’all are going to be our guiding light to help us make sure that all the functionality we add helps **you** serve the needs of your business. So if your developer experiences with the Semantic Layer match what we’ve written in this post, and especially if they don’t, I welcome [comments and feedback in this Discourse Discussion](https://discourse.getdbt.com/t/how-to-design-and-structure-metrics-the-discussion/4965)! Additionally, I recommend that you join us over at #dbt-core-metrics on the dbt Slack where I’ll be posting updates, answering questions, discussing usage, and hopefully responding with the best emojis.
```
Comment (jasnonaz) [inline website/blog/2022-10-18-how-to-design-metrics.md]:
```SUGGESTION 
This semantic layer is a fundamental change to what it means to interact with dbt and ultimately most of the best practices will come from the dbt Community - folks like you. It does not matter if you consider yourself an "expert" on this - we want to talk to you and hear how you are using or would like to use metrics and the semantic layer.  Y’all are going to be our guiding light to help us make sure that all the functionality we add helps **you** serve the needs of your business. 

If your experience with the Semantic Layer match what we’ve written in this post, and especially if they don’t, please share [comments and feedback in this Discourse Discussion](https://discourse.getdbt.com/t/how-to-design-and-structure-metrics-the-discussion/4965)! 

Additionally, I would invite you to join us over at #dbt-core-metrics on the dbt Slack where we’ll be posting updates, answering questions, discussing usage, and hopefully responding with the best emojis.

File: website/blog/2022-10-18-how-to-design-metrics.md [hunk 14]
```diff
@@ -0,0 +1,367 @@
...
+
+</details>
+
+**Is this what I should do?**
+
+It depends! There will be times when it might be better to have logic stored in the modeling layer and there will be times when it might be better to have logic stored in the semantic layer. The needs of your business stakeholders should drive your decision on where to keep this logic.
+
+## How to structure your metrics
+
+Now that we’ve designed our metrics, we move on to structuring them within our project. There are different ways to organize your metrics definitions and each comes with pros and cons.
```
Comment (jasnonaz) [inline website/blog/2022-10-18-how-to-design-metrics.md]:
```SUGGESTION 
Now that we’ve designed our metrics, let's move on to structuring them within our project. We'll examine the different ways to organize metrics  and take a look at the pros and cons of several strategies.

File: website/blog/2022-10-18-how-to-design-metrics.md [hunk 15]
```diff
@@ -0,0 +1,367 @@
...
+
+But that only gets us to the `promoter_pct` and `detractor_pct` metrics. In order to combine these both into a single metric definition, we needed to change the value that we assign. Given that the total range of values that the metric could output is -100 (all detractors) to 100 (all promoters) we can assign each of those categories that peak value, along with 0 for passives. This means that when the numerator is aggregated, it nets out promoters against detractors just like the documented equation does `promoter score - detractor score` . 
+
+</details>
+
+**Is this what I should do?**
+
+It depends! There will be times when it might be better to have logic stored in the modeling layer and there will be times when it might be better to have logic stored in the semantic layer. The needs of your business stakeholders should drive your decision on where to keep this logic.
```
Comment (jasnonaz) [inline website/blog/2022-10-18-how-to-design-metrics.md]: 1. Should we link this off of the "it depends" https://twitter.com/SBinLondon/status/1413113782214266885
2. Can we give some flavor to how folks might choose between the "it depends"

File: website/blog/2022-10-18-how-to-design-metrics.md [hunk 16]
```diff
@@ -0,0 +1,367 @@
...
+
+> Thoughtful curation of dimensions is essential for three main reasons:
+- ****Relevance:**** Analysts must include the dimensions most relevant to answering the question.
+- ****Trust****: Curating high-quality dimensions with little to no known errors helps ensure trust in analysis results and the decisions that follow.
+- ****Efficiency****: Curation provides a faster path to high-quality analysis results.
+> 
+
+To put it another way, **metrics are most useful when every dimension provided can help provide answers to the business.** 
+
+### Advanced example - how to design complicated metrics
```
Comment (jasnonaz) [inline website/blog/2022-10-18-how-to-design-metrics.md]: We might want to call this "complex" rather than "complicated" throughout

File: website/blog/2022-10-18-how-to-design-metrics.md [hunk 17]
```diff
@@ -0,0 +1,367 @@
...
+
+Now let’s look at a more complicated example of a metric that is calculated from components that could be metrics unto themselves. The metric in question is Net Promoter score, which is used by the dbt Labs internal analytics team to understand the experience that users are having on dbt Cloud.
```
Comment (jasnonaz) [inline website/blog/2022-10-18-how-to-design-metrics.md]:
```SUGGESTION
Now let’s look at a more complex example of a metric  - one that is built from components that could theoretically themselves be metrics. The metric in question is Net Promoter score, which is used by the dbt Labs internal analytics team to understand the experience that users are having on dbt Cloud.

Comment (jasnonaz) [reply website/blog/2022-10-18-how-to-design-metrics.md]: Also this may or may not be too much of a throwback but 
image

File: website/blog/2022-10-18-how-to-design-metrics.md [hunk 18]
```diff
@@ -0,0 +1,367 @@
...
+| feedback_date | unique_id | feedback_source | user_type | account_plan | score | nps_category |
+| --- | --- | --- | --- | --- | --- | --- |
+| 2022-10-01 | 1 | nps_tool_1 | developer | team | 5 | detractor |
+| 2022-10-01 | 2 | nps_tool_2 | read_only | developer | 8 | promoter |
+| 2022-10-02 | 3 | nps_tool_1 | developer | enterprise | 10 | promoter |
+| 2022-10-02 | 4 | nps_tool_1 | developer | developer | 7 | passive |
+| 2022-10-02 | 5 | nps_tool_2 | developer | team | 9 | promoter |
+| 2022-10-03 | 6 | nps_tool_1 | developer | enterprise | 7 | passive |
+
+The dbt Internal Analytics team ([long may they reign](https://www.linkedin.com/feed/update/urn:li:activity:6962884130569080833/)) took this data and decided to build the NPS Score metric into our repo so that it could be surfaced to stakeholders in multiple tools. This process is where we began to form our opinions on what should live in the modeling layer vs semantic layer.
```
Comment (jasnonaz) [inline website/blog/2022-10-18-how-to-design-metrics.md]: I know I have said that we caveated in too many places, but I think that a caveat might be useful here along the lines of. "This process is where we began to form our early opinions on what should live in the modeling layer vs semantic layer - but these are sure to progress as we add in more and more real world use cases."

File: website/blog/2022-10-18-how-to-design-metrics.md [hunk 19]
```diff
@@ -0,0 +1,367 @@
...
+tags: [dbt product updates]
+hide_table_of_contents: false
+
+date: 2022-10-18
+is_featured: true
+---
+
+---
+
+**IMPORTANT:** This document serves as the temporary location for information on how to design and structure your metrics. It is our intention to take this content and turn it into a Guide, similar to [How we structure our dbt projects](https://docs.getdbt.com/guides/best-practices/how-we-structure/1-guide-overview) , but we feel that codifying information in a Guide first requires that Metrics be rigorously tested by the community so that best practices can arise. This document contains our early attempts to create best practices. In other words, read these as suggestions for a new paradigm and share in the community where they do (or don’t) match your experiences! More information on where to do this can be found at the end.
```
Comment (mirnawong1) [inline website/blog/2022-10-18-how-to-design-metrics.md]: switch active voice
```suggestion
**IMPORTANT:** This document serves as the temporary location for information on how to design and structure your metrics. It is our intention to take this content and turn it into a Guide, like [How we structure our dbt projects](https://docs.getdbt.com/guides/best-practices/how-we-structure/1-guide-overview), but we feel that codifying information in a Guide first requires that Metrics be rigorously tested by the community so that best practices can arise. This document contains our early attempts to create best practices. In other words, read these as suggestions for a new paradigm and share in the community where they do (or don’t) match your experiences! You can find more information on where to do this at the end.
```

File: website/blog/2022-10-18-how-to-design-metrics.md (hunk 16 above)
Comment (mirnawong1) [reply website/blog/2022-10-18-how-to-design-metrics.md]: that's a great suggestion! complex or advanced is a good partner with 'basic'

Comment (mirnawong1) [inline website/blog/2022-10-18-how-to-design-metrics.md]:
```suggestion
### Advanced example - how to design complex metrics
```

File: website/blog/2022-10-18-how-to-design-metrics.md [hunk 20]
```diff
@@ -0,0 +1,367 @@
...
+
+Now let’s look at a more complicated example of a metric that is calculated from components that could be metrics unto themselves. The metric in question is Net Promoter score, which is used by the dbt Labs internal analytics team to understand the experience that users are having on dbt Cloud.
+
+For those of you who are unfamiliar with the industry metric of Net Promoter Score, here is a [great article from the folks over at Delighted on how it is calculated.](https://delighted.com/net-promoter-score) The short version of it is `the percentage of promoters - the percentage of detractors`. 
+
+---
+
+Here at dbt Labs we provide users with short surveys where they can provide feedback (as well as in a few other locations). The data is collected from those surveys is used to calculate our NPS Score, which helps us understand user sentiment over time.
+
+Given that these surveys come from a few different sources, there is a substantial amount of upstream modeling performed in order to unify them in a single model, but the end result is something that looks similar to the table below:
```
Comment (mirnawong1) [inline website/blog/2022-10-18-how-to-design-metrics.md]:
```suggestion
Given that these surveys come from a few different sources, there is a large amount of upstream modeling performed in order to unify them in a single model, but the end result is something that looks like the table below:
```

File: website/blog/2022-10-18-how-to-design-metrics.md (hunk 4 above)
Comment (callum-mcdata) [reply website/blog/2022-10-18-how-to-design-metrics.md]: Don't love it but added something

File: website/blog/2022-10-18-how-to-design-metrics.md (hunk 6 above)
Comment (callum-mcdata) [reply website/blog/2022-10-18-how-to-design-metrics.md]: made some changes but kept the first part because I like the flow of a hard cut after "I'll own that title!"

**Comment by mirnawong1** [comment]
hey @callum-mcdata this was great and i added some suggestions! i think @jasnonaz has added some more feedback, so i haven't approved it yet but @jasnonaz  let me know if you'd prefer to approve it!

File: website/blog/2022-10-18-how-to-design-metrics.md (hunk 5 above)
Comment (callum-mcdata) [reply website/blog/2022-10-18-how-to-design-metrics.md]: I mean its a bit of a historical distortion but if I had to pin down a single moment it was actually probably the "dbt should know about metrics" PR that opened my eyes to dbt being more than it was.

File: website/blog/2022-10-18-how-to-design-metrics.md (hunk 15 above)
Comment (callum-mcdata) [reply website/blog/2022-10-18-how-to-design-metrics.md]: 1. Yea!
2. I honestly don't know if we/me yet have opinions on when to do each option. The best short hand I've gotten has been "Put it all in the semantic layer if you want to analyze/use the individual components as well"

File: website/blog/2022-10-18-how-to-design-metrics.md (hunk 11 above)
Comment (callum-mcdata) [reply website/blog/2022-10-18-how-to-design-metrics.md]: Added

File: website/blog/2022-10-18-how-to-design-metrics.md (hunk 15 above)
Comment (callum-mcdata) [reply website/blog/2022-10-18-how-to-design-metrics.md]: Added a comment here

**Comment by jasnonaz** [comment]
Very, very happy with how this piece is coming together - excellent work. I just added a few (mostly minor wording) comments and suggestions. One final piece I think we can now add - the design and structure component of this is a really strong structure. Can we include, somewhere in the intro that this post will walk you through designing and strcuturing. Something like:

"This post will serve as our attempt to formulate best practices for metrics in two key areas:
- **Design**: What logic goes in metrics and how to apply calculations, filters and dimensions
- **Structure**: Where these will live in your dbt project and how to compose the files that contain your metrics"
That, combined with the new H2/H3 selection should fully lock in scannableness / readibility.

**Comment by callum-mcdata** [comment]
@jasnonaz I made the updates if you want to take a final pass! I think it should be good to go!

**Comment by jasnonaz** [comment]
Potential title ideas:
- How to design and structure dbt metrics: moving towards a semantic dbt
- How to design and structure dbt metrics: updating our guidance for a semantic dbt
- How to design and structure dbt metrics: recommendations for how to get started building semantically
- How to design and structure dbt metrics: recommendations for getting started

File: website/blog/2022-10-18-how-to-design-and-structure-metrics.md [hunk 21]
```diff
@@ -0,0 +1,381 @@
+---
+title: "How to design and structure dbt metrics: recommendations for getting started"
```
Comment (KiraFuruichi) [inline website/blog/2022-10-18-how-to-design-and-structure-metrics.md]:
```suggestion
title: "How to design and structure dbt metrics: Recommendations for getting started"
```

File: website/blog/2022-10-18-how-to-design-and-structure-metrics.md [hunk 22]
```diff
@@ -0,0 +1,381 @@
...
+
+---
+
+## The power of a semantic layer on top of a mature data modeling framework
+
+As a longtime dbt Community member, I knew I had to get involved when I first saw the dbt Semantic Layer in the now infamous [`dbt should know about metrics` Github Issue](https://github.com/dbt-labs/dbt-core/issues/4071). It gave me a vision of a world where metrics and business logic were unified across an entire organization; a world where the data team was no longer bound to a single consuming experience and could enable their stakeholders in dozens of different ways. To me, it felt like the opportunity to contribute to the next step of what dbt could become.
```
Comment (KiraFuruichi) [inline website/blog/2022-10-18-how-to-design-and-structure-metrics.md]:
```suggestion
As a longtime [dbt Community](https://www.getdbt.com/community/join-the-community/) member, I knew I had to get involved when I first saw the dbt Semantic Layer in the now infamous [`dbt should know about metrics` Github Issue](https://github.com/dbt-labs/dbt-core/issues/4071). It gave me a vision of a world where metrics and business logic were unified across an entire organization; a world where the data team was no longer bound to a single consuming experience and could enable their stakeholders in dozens of different ways. To me, it felt like the opportunity to contribute to the next step of what dbt could become.
```

File: website/blog/2022-10-18-how-to-design-and-structure-metrics.md [hunk 23]
```diff
@@ -0,0 +1,381 @@
...
+
+In past roles, I’ve been referred to as the `dbt zealot` and I’ll gladly own that title! It’s not a surprise - dbt was built to serve data practicioners expand the power of our work with software engineering principles. It gave us flexibility and power to serve our organizations. But I always wondered if there were more folks who could directly benefit from interacting with dbt.
+
+The Semantic Layer expands the reach of dbt **by coupling dbt’s mature data modeling framework with semantic definitions.** The result is a first of its kind data experience that serves both the data practicioners writing your analytics code and stakeholders who depend on it. Metrics are the first step towards this vision, allowing users to version control and centrally define their key business metrics in a single repo while also serving them to the entire business.
+
+However, this is still a relatively new part of the dbt toolbox and you probably have a lot of questions on how exactly you can do that. This blog contains our early best practice recommendations for metrics in two key areas:
+- **Design**: What logic goes into metrics and how to use calculations, filters, dimensions, etc
```
Comment (KiraFuruichi) [inline website/blog/2022-10-18-how-to-design-and-structure-metrics.md]:
```suggestion
- **Design**: What logic goes into metrics and how to use calculations, filters, dimensions, etc.
```

Add period

File: website/blog/2022-10-18-how-to-design-and-structure-metrics.md [hunk 24]
```diff
@@ -0,0 +1,381 @@
...
+
+Our instinct when designing metrics might be to encode as much information as possible into the semantic layer. An example of this is case statements - the analytics engineer’s gut instinct might be to mimic tools of the past and provide complicated case statements for the metric `expression` property to try and capture the nuance of how it should be calculated. 
+
+But remember - you always have the option of performing this logic _in the modeling layer_. This is the key difference between dbt and other semantic layer offerings - by sitting the semantic layer atop a mature transformation layer, you always have the option to configure and optimize your logic within your models and then _define semantic components with intentionality_. 
+
+Getting the balance  just right is a learning experience and developing community best practices and standards will take time, which is why it’s important for us to think from first principles. What should really be our goal when determining whether logic lives in a model or a metrics? 
+
+To explore this question and begin to develop an intuition, we’ll walk through two examples of handling this divide.
+
+## Basic example - revenue
```
Comment (KiraFuruichi) [inline website/blog/2022-10-18-how-to-design-and-structure-metrics.md]:
```suggestion
## Basic example: Revenue
```

File: website/blog/2022-10-18-how-to-design-and-structure-metrics.md [hunk 25]
```diff
@@ -0,0 +1,381 @@
...
+
+In plain english, the `expression` property is the sql column (or expression) that we are applying the calculation method on. In our example above, this simplifies to `sum(amount)`. However, this doesn’t **need** to be a field in the model. It could also be a sql expression like `case when condition = true then field else 0 end` .
+
+And then there’s `dimensions`.
+
+### Choosing which dimensions to use with your metric
+
+The `dimensions` attribute is a bit more nuanced than the others because it involves curating the ways through which a user can interact with the metric. To that end …
+
+*❗ **We recommend curating dimensions, not including all columns within the model. Most models contain dimensions that aren’t relevant for end-user analysis.***
```
Comment (KiraFuruichi) [inline website/blog/2022-10-18-how-to-design-and-structure-metrics.md]: There's something wonky happening here with the asterisks again, looks like there's an extra one hanging out in the beginning?

File: website/blog/2022-10-18-how-to-design-and-structure-metrics.md [hunk 26]
```diff
@@ -0,0 +1,381 @@
...
+
+> Thoughtful curation of dimensions is essential for three main reasons:
+- **Relevance:** Analysts must include the dimensions most relevant to answering the question.
+- **Trust**: Curating high-quality dimensions with little to no known errors helps ensure trust in analysis results and the decisions that follow.
+- **Efficiency**: Curation provides a faster path to high-quality analysis results.
+> 
+
+To put it another way, **metrics are most useful when every dimension provided can help provide answers to the business.** 
+
+## Advanced example - NPS
```
Comment (KiraFuruichi) [inline website/blog/2022-10-18-how-to-design-and-structure-metrics.md]:
```suggestion
## Advanced example: NPS
```

File: website/blog/2022-10-18-how-to-design-and-structure-metrics.md [hunk 27]
```diff
@@ -0,0 +1,381 @@
...
+| 2022-10-01 | 1 | nps_tool_1 | developer | team | 5 | detractor |
+| 2022-10-01 | 2 | nps_tool_2 | read_only | developer | 8 | promoter |
+| 2022-10-02 | 3 | nps_tool_1 | developer | enterprise | 10 | promoter |
+| 2022-10-02 | 4 | nps_tool_1 | developer | developer | 7 | passive |
+| 2022-10-02 | 5 | nps_tool_2 | developer | team | 9 | promoter |
+| 2022-10-03 | 6 | nps_tool_1 | developer | enterprise | 7 | passive |
+
+The dbt Internal Analytics team ([long may they reign](https://www.linkedin.com/feed/update/urn:li:activity:6962884130569080833/)) took this data and decided to build the NPS Score metric into our repo so that it could be surfaced to stakeholders in multiple tools. This process is where we began to form our opinions on what should live in the modeling layer vs semantic layer - but these are sure to progress as we add in more and more real world use cases.
+
+### Option 1 - putting everything in the semantic layer
```
Comment (KiraFuruichi) [inline website/blog/2022-10-18-how-to-design-and-structure-metrics.md]:
```suggestion
### Option 1: Putting everything in the semantic layer
```

File: website/blog/2022-10-18-how-to-design-and-structure-metrics.md [hunk 28]
```diff
@@ -0,0 +1,381 @@
...
+	    timestamp: created_at
+	    time_grains: [day, month, quarter, year]
+	    dimensions:
+	      - feedback_source
+	      - account_plan
+	      - user_type
+
+```
+
+### Option 2 - keeping logic in the modeling layer
```
Comment (KiraFuruichi) [inline website/blog/2022-10-18-how-to-design-and-structure-metrics.md]:
```suggestion
### Option 2: Keeping logic in the modeling layer
```

**Review by KiraFuruichi — APPROVED** [review]
This is such a well-written and thorough blog post!!

I left some minor aesthetic comments, but don't think they're things that block this article, so giving the ✅ from me!!
